src/darwin/Framework/CHIP/MTRDeviceController.h

```diff
@@ -39,61 +41,60 @@ typedef void (^MTRDeviceConnectionCallback)(MTRBaseDevice * _Nullable device, NS
 @property (readonly, nonatomic, nullable) NSNumber * controllerNodeID;
 
 /**
- * Start pairing for a device with the given ID, using the provided setup PIN
- * to establish a PASE connection.
+ * Set up a commissioning session for a device, using the provided setup payload
+ * to discover it and connect to it.
+ *
+ * @param payload a setup payload (probably created from a QR code or numeric
+ *                code onboarding payload).
+ * @param newNodeID the planned node id for the node.
+ * @error error indication if discovery can't start at all (e.g. because the
+ *              setup payload is invalid).
  *
  * The IP and port for the device will be discovered automatically based on the
  * provided discriminator.
  *
- * The pairing process will proceed until a PASE session is established or an
- * error occurs, then notify onPairingComplete on the MTRDevicePairingDelegate
- * for this controller.  That delegate is expected to call commissionDevice
- * after that point if it wants to commission the device.
- */
-- (BOOL)pairDevice:(uint64_t)deviceID
-     discriminator:(uint16_t)discriminator
-      setupPINCode:(uint32_t)setupPINCode
-             error:(NSError * __autoreleasing *)error;
-
-/**
- * Start pairing for a device with the given ID, using the provided IP address
- * and port to connect to the device and the provided setup PIN to establish a
- * PASE connection.
- *
- * The pairing process will proceed until a PASE session is established or an
- * error occurs, then notify onPairingComplete on the MTRDevicePairingDelegate
- * for this controller.  That delegate is expected to call commissionDevice
- * after that point if it wants to commission the device.
+ * Then a PASE session will be established with the device, unless an error
+ * occurs.  MTRDevicePairingDelegate will be notified as follows:
+ *
+ * * Discovery fails: onStatusUpdate with MTRPairingStatusFailed.
+ *
+ * * Discovery succeeds but commissioning session setup fails: onPairingComplete
+ *   with an error.
+ *
+ * * Commissioning session setup succeeds: onPairingComplete with no error.
+ *
+ * Once a commissioning session is set up, deviceBeingCommissionedWithNodeID
+ * can be used to get an MTRBaseDevice and discover what sort of network
+ * credentials the device might need, and commissionNodeWithID can be used to
+ * commission the device.
  */
-- (BOOL)pairDevice:(uint64_t)deviceID
-           address:(NSString *)address
-              port:(uint16_t)port
-      setupPINCode:(uint32_t)setupPINCode
-             error:(NSError * __autoreleasing *)error;
+- (void)setupCommissioningSessionWithPayload:(MTRSetupPayload *)payload
```

Comment:
This change would benefit greatly the overall CHIPDeviceController API

Reply:
Yes, that is the idea.  :)

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/22603